File: apps/dashboard-service/src/main/java/com/github/mankit007/recipe/Dashboard/Service/model/ClickEvent.java
package com.github.mankit007.recipe.Dashboard.Service.model;

import java.time.Instant;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "click_events")
public class ClickEvent {

    @Id
    private String id;

    @Field("short_code")
    private String shortCode;

    @Field("original_url")
    private String originalUrl;

    @Field("tenant_id")
    private String tenantId;

    @Field("timestamp")
    private Instant timestamp;

    @Field("user_agent")
    private String userAgent;

    @Field("referer")
    private String referer;

    @Field("ip_address")
    private String ipAddress;
}

File: apps/redirector-service/model/click_event.go
package model

import "time"

type ClickEvent struct {
	ShortCode   string    `json:"short_code"`
	OriginalURL string    `json:"original_url"`
	TenantID    string    `json:"tenant_id"`
	Timestamp   time.Time `json:"timestamp"`
	UserAgent   string    `json:"user_agent"`
	Referer     string    `json:"referer"`
	IPAddress   string    `json:"ip_address"`
}
